src/builtins/echo.rs

```diff
@@ -0,0 +1,146 @@
+use std::io::{self, Write, BufWriter};
+
+const HELP: u8 = 1;
+const ESCAPE: u8 = 2;
+const NO_NEWLINE: u8 = 3;
+const NO_SPACES: u8 = 4;
+
+const MAN_PAGE: &'static str = /* @MANSTART{echo} */ r#"
+NAME
+    echo - display a line of text
+SYNOPSIS
+    echo [ -h | --help ] [-e] [-n] [-s] [STRING]...
+DESCRIPTION
+    Print the STRING(s) to standard output.
+OPTIONS
+    -e
+        enable the interpretation of backslash escapes
+    -n
+        do not output the trailing newline
+    -s
+        do not separate arguments with spaces
+    Escape Sequences
+        When the -e argument is used, the following sequences will be interpreted:
+        \\  backslash
+        \a  alert (BEL)
+        \b  backspace (BS)
+        \c  produce no further output
+        \e  escape (ESC)
+        \f  form feed (FF)
+        \n  new line
+        \r  carriage return
+        \t  horizontal tab (HT)
+        \v  vertical tab (VT)
+"#; /* @MANEND */
+
+pub fn echo(args: &[String]) -> Result<(), io::Error> {
+    let mut flags = 0u8;
+    let mut data: Vec<&str> = vec![];
+
+    for arg in args {
+        match arg.as_str() {
+            "--help" => flags |= HELP,
+            "--escape" => flags |= ESCAPE,
+            "--no-newline" => flags |= NO_NEWLINE,
+            "--no-spaces" => flags |= NO_SPACES,
+            _ => {
+                if arg.starts_with("-") {
+                    let arg = &arg[1..];
+                    for argopt in arg.chars() {
+                        match argopt {
+                            'e' => flags |= ESCAPE,
+                            'n' => flags |= NO_NEWLINE,
+                            's' => flags |= NO_SPACES,
+                            'h' => flags |= HELP,
+                            _ => (),
+                        }
+                    }
+                } else {
+                    data.push(arg);
+                }
+            }
+        }
+    }
+
+    let stdout = io::stdout();
+    let mut buffer = BufWriter::new(stdout.lock());
+
+    if (flags & HELP) != 0 {
```

Comment:
Bit flags are checked by comparing the AND of the reversed bits of the flag, so `if flags & (255 ^ HELP) != 0`.